collectors/0/samza.py

```diff
@@ -64,10 +61,41 @@ def main(argv):
         metrics_raw = message_json['metrics']
         header_raw = message_json['header']
 
-        metrics = {}
-        for m in METRICS_TO_REPORT:
-            if m in  metrics_raw:
-                metrics[m] = metrics_raw[m]
+        # reporting logic is specific to the type of metric reported
+        report_jvm_and_container_metrics(metrics_raw, header_raw)
+        report_consumer_lag(metrics_raw, header_raw)
+
+
+def report_jvm_and_container_metrics(metrics_raw, header_raw):
+
+    metrics = {}
+    for m in ['org.apache.samza.metrics.JvmMetrics', 'org.apache.samza.container.SamzaContainerMetrics']:
```

Comment:
Yes, it was cleaner before. However, since we are adding special logic to extract more specific metrics, it is not possible to simply whitelist the top level metric names.